### PR TITLE
Update head-object api

### PR DESCRIPTION
**Pull request by priyanka25081999**
- Add getter methods for response header elements of head-object s3 api.

Signed-off-by: Priyanka Salunke <priyanka.s.salunke@seagate.com>

### CODE REVIEW

File: s3/replication/common/src/s3replicationcommon/s3_head_object.py [hunk 1]
```diff
@@ -90,12 +436,12 @@ async def get(self):
                     headers=headers) as resp:
 
                 if resp.status == 200:
-                    self._response_headers = resp.headers
+                    self._response_headers = dict(resp.headers)
                     self._logger.info(
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/s3_head_object.py:440]: manually format and reduce the number of lines used by this statement and check for other places too.

Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_head_object.py]: done

File: s3/replication/common/src/s3replicationcommon/s3_head_object.py [hunk 2]
```diff
@@ -42,9 +42,357 @@ def __init__(self, session, request_id,
...
         self._state = S3RequestState.INITIALISED
 
-    def get_contentlength(self):
-        """Returns content length for object."""
-        return int(self._response_headers["Content-Length"])
+    def get_accept_ranges(self):
+        """Get range of bytes for object.
+
+        Returns
+        -------
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/s3_head_object.py:49]: please use google format of doc string. Reference : https://realpython.com/documenting-python-code/#google-docstrings-example

applicable to this complete file.

File: s3/replication/common/src/s3replicationcommon/s3_head_object.py [hunk 3]
```diff
@@ -42,9 +42,357 @@ def __init__(self, session, request_id,
...
+        if self._resp_parts_count is not None:
+            self._resp_parts_count = int(self._resp_parts_count)
+        return self._resp_parts_count
+
+    def get_x_amz_object_lock_legal_hold(self):
+        """Get legal hold status value for the object.
+
+        Returns
+        -------
+            str: ON if a legal hold is in effect for the object, else OFF.
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/s3_head_object.py:235]: will it be capital response? or you have put it like this to emphasize?

File: s3/replication/common/src/s3replicationcommon/s3_head_object.py (hunk 2 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_head_object.py:49]: Google format of docstring is not working. Codacy is showing an error for google docstrings. So I have kept the original docstring format.

File: s3/replication/common/src/s3replicationcommon/s3_head_object.py (hunk 3 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_head_object.py]: yes, it will be a capital response, having return value either ON/OFF.

**Comment by rajkumarpatel2602** [comment]
Looks good to me.

**Review by rajkumarpatel2602 — APPROVED** [review]

**Pull request merged**